### ISSUE TITLE
Views dropbutton wrongly handles preset link destination

### ISSUE THREAD

**Issue by argiepiano** [backdrop/backdrop-issues#5676]
## Description of the bug

This is a rewrite (for clarity) of the closed #5675.

Views' dropbutton field wrongly handles the preset "destination" portion of its links.

## Steps To Reproduce

1. Create a test view of Content, using a table. In "page settings" add the path admin/testview
2. Add the field Content: Nid 
3. Add a Custom text field, and type "link" in its "Text" box
4. In "Rewrite results" select "Output this field as a link" and type the path "node/[nid]/edit?destination=node/[nid]"
5. Add a dropbutton and select the Custom text. Be sure "Include destination" is **unchecked**
6. Save the view. Visit the view. See snapshot. 
7. If you hover over the links, you'll see the the correct link and correct destination. 👍🏽 
8. But if you hover over the dropbutton, you will see `http://example.com/node/6/edit%3Fdestination%3Dnode/6` 👎🏽 

<img width="1054" alt="Screen Shot 2022-07-06 at 4 45 08 PM" src="https://user-images.githubusercontent.com/9938978/177655520-42988c61-8b9e-4d84-a316-0a81e015af48.png">

The dropbutton link above doesn't take you to the correct place (node edit form), nor does the destination work at all

Now, for more fun:

1. Go back to the view and edit the dropbutton field
2. Be sure to check "Include destination" and save
3. Go to the view
4. The dropbutton link now looks even worse: `http://example.com/node/6/edit%3Fdestination%3Dnode/6?destination=admin/testview`

The dropbutton appended a destination (basically the current page), and mangled the original link destination.

## Actual behavior

See above

## Expected behavior

Dropbutton should respect the destination of the link if set:
1. If the link has a destination, it should include it properly formatted
2. If the link doesn't have a destination, it should append the current page as destination when "Include destination" is checked. 

## Additional information

- Backdrop CMS version: 1.22.0

**Comment by indigoxela** [comment]
Maybe there's a misunderstanding, what "Include destination" does?

Help text:
Include a "destination" parameter in the link to return the user **to the original view upon completing the action**.

So if you want to set a *custom* destination, that setting's not what you want.

Now for the mangling: "Output this field as a link" might be the culprit, as it seems to urlencode. I'm not sure if that's a bug, though. IMO it''s not the dropbutton, but the link field, that mangles the url. Might need a little more digging.

Field type "Global dropbutton" seems a bit picky, which fields can be used. We might be able to improve that.

**Comment by argiepiano** [comment]
> Include a "destination" parameter in the link to return the user to the original view upon completing the action.

OK, fair point. Did not catch that. I've been able to do what I describe without any problems in D7 and I expected backward compatibility.

> "Output this field as a link" might be the culprit
No, it's not. The "link" (which is a Custom text output as a link) by itself doesn't get mangled. It's only when you include it as part of a dropbutton. The culprit handler is `views_handler_field_links`, which is only used by `views_handler_field_dropbutton` (in fact the latter extends the former - no other views field uses `views_handler_field_dropbutton`. The Custom text field handles links internally).

**Comment by argiepiano** [comment]
No, thank YOU for all your work around the community and taking an interest on this topic. You are amazing! 

It will probably take me a few days but I intend to file a PR here; if you have a chance it'd be great if you could take a look.

**Comment by indigoxela** [comment]
> It will probably take me a few days but I intend to file a PR here;

Great! Ping me for testing.

For completeness: views_handler_field_dropbutton doesn't exist in D7 Views. It came in with #143.

**Comment by argiepiano** [comment]
Thanks. Will take a look.

The dropbutton functionality and a handler did indeed exist in D7's Views. It was called `views_handler_field_ctools_dropdown` and you can find it [here](https://git.drupalcode.org/project/views/-/blob/7.x-3.x/handlers/views_handler_field_ctools_dropdown.inc). Its code differs somewhat from the current Backdrop version.
